packages/@aws-cdk/aws-glue/lib/security-configuration.ts

```diff
@@ -0,0 +1,239 @@
+}
+
+/**
+ * S3 encryption configuration.
+ */
+export interface S3Encryption {
+  /**
+   * Encryption mode.
+   */
+  readonly mode: S3EncryptionMode,
+
+  /**
+   * The KMS key to be used to encrypt the data.
+   * @default no kms key if mode = DISABLED or S3_MANAGED. A key must be provided for KMS mode.
+   */
+  readonly kmsKey?: kms.IKey,
+}
+
+/**
+ * CloudWatch Logs encryption configuration.
+ */
+export interface CloudWatchEncryption {
+  /**
+   * Encryption mode
+   */
+  readonly mode: CloudWatchEncryptionMode;
+
+  /**
+   * The KMS key to be used to encrypt the data.
+   * @default no kms key if mode = DISABLED. A key must be provided otherwise.
+   */
+  readonly kmsKey?: kms.IKey,
+}
+
+/**
+ * Job bookmarks encryption configuration.
+ */
+export interface JobBookmarksEncryption {
+  /**
+   * Encryption mode.
+   */
+  readonly mode: JobBookmarksEncryptionMode;
+
+  /**
+   * The KMS key to be used to encrypt the data.
+   * @default no kms key if mode = DISABLED. A key must be provided otherwise.
+   */
+  readonly kmsKey?: kms.IKey,
+}
+
+/**
+ * Constructions properties of {@link SecurityConfiguration}.
+ */
+export interface SecurityConfigurationProps {
+  /**
+   * The name of the security configuration.
+   */
+  readonly securityConfigurationName: string;
+
+  /**
+   * The encryption configuration for Amazon CloudWatch Logs.
+   * @default no cloudwatch logs encryption.
+   */
+  readonly cloudWatchEncryption?: CloudWatchEncryption,
+
+  /**
+   * The encryption configuration for Glue Job Bookmarks.
+   * @default no job bookmarks encryption.
+   */
+  readonly jobBookmarksEncryption?: JobBookmarksEncryption,
+
+  /**
+   * The encryption configuration for Amazon Simple Storage Service (Amazon S3) data.
+   * @default no s3 encryption.
+   */
+  readonly s3Encryption?: S3Encryption,
+}
+
+/**
+ * A security configuration is a set of security properties that can be used by AWS Glue to encrypt data at rest.
+ *
+ * The following scenarios show some of the ways that you can use a security configuration.
+ * - Attach a security configuration to an AWS Glue crawler to write encrypted Amazon CloudWatch Logs.
+ * - Attach a security configuration to an extract, transform, and load (ETL) job to write encrypted Amazon Simple Storage Service (Amazon S3) targets and encrypted CloudWatch Logs.
+ * - Attach a security configuration to an ETL job to write its jobs bookmarks as encrypted Amazon S3 data.
+ * - Attach a security configuration to a development endpoint to write encrypted Amazon S3 targets.
+ */
+export class SecurityConfiguration extends cdk.Resource implements ISecurityConfiguration {
+
+  /**
+   * Creates a Connection construct that represents an external connection.
+   *
+   * @param scope The scope creating construct (usually `this`).
+   * @param id The construct's id.
+   * @param attrs Import attributes.
+   */
+  public static fromSecurityConfigurationAttributes(scope: constructs.Construct, id: string,
```

Comment:
Lets just start with a `fromSecurityConfigurationName` - the attributes interface seems unnecessary at this point.

Reply:
done